src/overlays/actors/ovl_Boss_01/z_boss_01.c

```diff
@@ -246,191 +647,2935 @@ ActorInit Boss_01_InitVars = {
+        }
+    }
+
+    if (this->lookAtPlayer) {
+        targetHeadRotY = this->actor.yawTowardsPlayer - this->actor.shape.rot.y;
+        if ((targetHeadRotY > 0x3800) || (targetHeadRotY < -0x3800)) {
+            targetHeadRotY = 0;
+        }
+
+        diffX = player->actor.world.pos.x - this->actor.focus.pos.x;
+        yDiff = KREG(36) + ((player->actor.world.pos.y + 25.0f) - this->actor.focus.pos.y);
+        diffZ = player->actor.world.pos.z - this->actor.focus.pos.z;
+        targetHeadRotX = Math_Atan2S(yDiff, sqrtf(SQ(diffX) + SQ(diffZ)));
+
+        // This line of code ensures that *not* following the player is the default behavior; if you want Odolwa's head
+        // to track the player, you'll need to set this variable to true for every single frame you want this behavior.
+        this->lookAtPlayer = false;
+    } else {
+        targetHeadRotX = 0;
+        targetHeadRotY = 0;
+    }
+
+    Math_ApproachS(&this->headRotY, targetHeadRotY, 3, 0x2000);
+    Math_ApproachS(&this->headRotX, targetHeadRotX, 3, 0x2000);
+
+    DECR(this->drawDmgEffTimer);
+
+    switch (this->drawDmgEffState) {
+        case ODOLWA_DRAW_DMG_EFF_STATE_NONE:
+            this->drawDmgEffType = ACTOR_DRAW_DMGEFF_FIRE;
+            this->drawDmgEffTimer = 0;
+            this->drawDmgEffAlpha = 0.0f;
+            break;
+
+        case ODOLWA_DRAW_DMG_EFF_STATE_FIRE_INIT:
+            this->drawDmgEffAlpha = 1.0f;
+            this->drawDmgEffScale = 0.0f;
+            this->drawDmgEffType = ACTOR_DRAW_DMGEFF_FIRE;
+            this->drawDmgEffTimer = 40;
+            this->drawDmgEffState++;
+            Actor_SetColorFilter(&this->actor, COLORFILTER_COLORFLAG_RED, 120, COLORFILTER_BUFFLAG_OPA, 60);
+            // fallthrough
+
+        case ODOLWA_DRAW_DMG_EFF_STATE_FIRE_ACTIVE:
+            if (this->drawDmgEffTimer == 0) {
+                Math_ApproachZeroF(&this->drawDmgEffAlpha, 1.0f, 0.02f);
+                if (this->drawDmgEffAlpha == 0.0f) {
+                    this->drawDmgEffState = ODOLWA_DRAW_DMG_EFF_STATE_NONE;
+                }
+            } else {
+                Math_ApproachF(&this->drawDmgEffScale, 1.0f, 0.1f, 0.5f);
+            }
+            break;
+
+        case ODOLWA_DRAW_DMG_EFF_STATE_FROZEN_INIT:
+            this->drawDmgEffType = ACTOR_DRAW_DMGEFF_FROZEN_SFX;
+            this->drawDmgEffTimer = 40;
+            this->drawDmgEffState++;
+            this->drawDmgEffAlpha = 1.0f;
+            this->drawDmgEffScale = 0.0f;
+            this->drawDmgEffFrozenSteamScale = 1.0f;
+            // fallthrough
+
+        case ODOLWA_DRAW_DMG_EFF_STATE_FROZEN_ACTIVE:
+            if (this->drawDmgEffTimer == 0) {
+                Boss01_Thaw(this, play);
+                this->drawDmgEffState = ODOLWA_DRAW_DMG_EFF_STATE_NONE;
+                break;
+            } else if (this->drawDmgEffTimer == 50) {
+                this->drawDmgEffType = ACTOR_DRAW_DMGEFF_FROZEN_NO_SFX;
+            }
```

Comment:
```suggestion
            if (this->drawDmgEffTimer == 0) {
                Boss01_Thaw(this, play);
                this->drawDmgEffState = ODOLWA_DRAW_DMG_EFF_STATE_NONE;
                break;
            } 
            
            if (this->drawDmgEffTimer == 50) {
                this->drawDmgEffType = ACTOR_DRAW_DMGEFF_FROZEN_NO_SFX;
            }
```
Because of the break, can the if be separated?